app/frontend/components/shared/select/selectors/contact-select.tsx

```diff
@@ -76,26 +109,55 @@ export const ContactSelect = observer(({ onChange, selectedOption, stylesToMerge
       <Button onClick={handleCreate} leftIcon={<Plus />} variant="primary" w="28%" px={2}>
         {t("contact.create")}
       </Button>
-      <CreateContactModal
+      <CreateEditContactModal
         isOpen={isOpen}
-        onClose={onClose}
-        onCreate={(contact) => onChange({ label: "", value: contact })}
+        onClose={() => {
+          setEditingContact(null)
+          onClose()
+        }}
+        contact={editingContact}
+        onDestroy={onDestroy}
+        onCreateOrUpdate={onCreateOrUpdate}
       />
     </>
   )
 })
 
-const Option = (props: OptionProps<IOption<IContact>>) => {
-  const label = props.label
-  const value = props.data.value
+interface IOptionComponentProps extends OptionProps<IOption<IContact>> {
+  onClickEdit: (IContact) => void
+}
+
+const Option = ({ onClickEdit, ...props }: IOptionComponentProps) => {
+  const { t } = useTranslation()
+  const {
+    label,
+    data: { value },
+  } = props
```

Comment:
Did you test trying to pass a value select? Does this cause issue. I would assume it should since the destructuring shouldn't work. Just double check to make sure, and if it's working it's good to got

Reply:
this gets the value for the option, not sure what you mean as this is working fine